lib/active_shipping/shipment_packer.rb

```diff
@@ -40,6 +40,7 @@ def pack(items, dimensions, maximum_weight, currency)
             items_to_pack.reject! { |i| i[:quantity].to_i == 0 }
             state = :package_full
           when :package_full
+            validate_package_quantity(packages.count)
```

Comment:
hmm thats a good point, this can definitely be in `:filling_package`, as long as we're enforcing the upper bound at some point in the loop I'm okay with that.

I'll split this PR up